### ISSUE TITLE
flickering

### ISSUE THREAD

**Issue by reacocard** [jellysquid3/Sodium#778]
### Version information

sodium-fabric-mc1.17.1-0.3.0+build.5

### Expected Behavior

renders smoothly without flickering

### Actual Behavior

lots of flickering of surfaces, especially at medium-long range. see attached video

### Reproduction Steps

happens every time on my system, first noticed with ocean surface

### Java version

16.0.1 64bit

### CPU

intel i7-1165G7

### GPU

intel Xe

### Additional information

video of issue: https://user-images.githubusercontent.com/178923/126093737-18d6c09d-9ecb-4692-9b49-9f5a67914e9f.mp4

mods:
```
architectury-2.2.21-fabric.jar
cloth-config-5.0.34-fabric.jar
easierchests-1.17.1-fabric0.36.1-1.6.3.jar
fabric-api-0.37.0+1.17.jar
fabric-language-kotlin-1.6.2+kotlin.1.5.20.jar
fabric-phc2foodcore-1.0.1.jar
hwyla-addon-horseinfo-0.3.0-1.17.jar
malilib-fabric-1.17.0-0.10.0-dev.22+beta.1.jar
megane-5.1.0.jar
modmenu-2.0.2.jar
pal-1.3.0.jar
quickbind-1.1.jar
RoughlyEnoughItems-6.0.264-alpha-fabric.jar
sodium-fabric-mc1.17.1-0.3.0+build.5.jar
wthit-fabric-3.7.0.jar
Xaeros_Minimap_FP21.13.0_Fabric_1.17.1.jar
XaerosWorldMap_1.14.6.1_Fabric_1.17.1.jar
```

Running on Debian linux, mesa version 20.3.5, kernel version 5.10

**Comment by Altirix** [comment]
could you do a test with just sodium loaded for the devs just incase the flickering is due to mod incompatibility 

also looks like this could be related to https://github.com/CaffeineMC/sodium-fabric/issues/24 but not as broken

**Comment by reacocard** [comment]
Still happens with only sodium loaded. Also looked at latest.log while testing this time and there's no errors or warnings in it.

**Comment by reacocard** [comment]
I upgraded to mesa 21.1.4 from debian experimental and i don't get the flickering anymore! guess it got fixed sometime between mesa 20.3.5 and 21.1.4

edit: i see a bunch of commits around multi-draw in [the release notes for mesa 21.0.0](https://gitlab.freedesktop.org/mesa/mesa/-/blob/main/docs/relnotes/21.0.0.rst) so given #24 i'm guessing that's the release that fixed it.

**Comment by jellysquid3** [comment]
If you're confident that upgrading to Mesa 21.1.4 resolved the issue, then we'll go ahead and close this issue. Please re-open it if you encounter the problem again.
